modules/services/mapilio.js

```
@@ -0,0 +1,564 @@
+    limit = limit || 5;
+
+    return partitionViewport(projection)
+        .reduce(function(result, extent) {
+            const found = rtree.search(extent.bbox())
+                .slice(0, limit)
+                .map(function(d) { return d.data; });
+
+            return (found.length ? result.concat(found) : result);
+        }, []);
+}
+
+// Load all data for the specified type from Mapilio vector tiles
+function loadTiles(which, url, maxZoom, projection) {
+    const tiler = utilTiler().zoomExtent([minZoom, maxZoom]).skipNullIsland(true);
+    const tiles = tiler.getTiles(projection);
+
+    tiles.forEach(function(tile) {
+        loadTile(which, url, tile);
+    });
+}
+
+
+// Load all data for the specified type from one vector tile
+function loadTile(which, url, tile) {
+    const cache = _mlyCache.requests;
+    const tileId = `${tile.id}-${which}`;
+    if (cache.loaded[tileId] || cache.inflight[tileId]) return;
+    const controller = new AbortController();
+    cache.inflight[tileId] = controller;
+    const requestUrl = url.replace('{x}', tile.xyz[0])
+        .replace('{y}', tile.xyz[1])
+        .replace('{z}', tile.xyz[2]);
+
+    fetch(requestUrl, { signal: controller.signal })
+        .then(function(response) {
+            if (!response.ok) {
+                throw new Error(response.status + ' ' + response.statusText);
+            }
+            cache.loaded[tileId] = true;
+            delete cache.inflight[tileId];
+            return response.arrayBuffer();
+        })
+        .then(function(data) {
+            if (!data) {
+                throw new Error('No Data');
+            }
+
+            loadTileDataToCache(data, tile, which);
+
+            if (which === 'images') {
+                dispatch.call('loadedImages');
+            } else {
+                dispatch.call('loadedLines');
+            }
+        })
+        .catch(function() {
+            cache.loaded[tileId] = true;
```

This is a bit problematic, as it silently ignores all kinds of exceptions, not just the "No data" case , I assume you want to handle here.

I'd say that for generic errors (e.g. a non-HTTP/200 response from the server, corrupted data which cannot be parsed, or just javascript runtime errors), there should be at least a console warning output, maybe like this:

```suggestion
        .catch(function(e) {
            if (e.message === 'No Data') {
                cache.loaded[tileId] = true;
            } else {
                console.error(e);
            }
```